pkg/manager/impl/execution_manager.go

```diff
@@ -1343,6 +1346,16 @@ func (m *ExecutionManager) CreateWorkflowEvent(ctx context.Context, request admi
 			go m.emitScheduledWorkflowMetrics(ctx, executionModel, request.Event.OccurredAt)
 		}
 	} else if common.IsExecutionTerminal(request.Event.Phase) {
+		if request.Event.Phase == core.WorkflowExecution_FAILED {
+			// request.Event is expected to be of type WorkflowExecutionEvent_Error when workflow fails.
+			// if not, log the error and continue
+			if err := request.Event.GetError(); err != nil {
+				ctx = context.WithValue(ctx, errorKindKey, err.Kind.String())
```

Comment:
I think you may need to update the existing call [SetMetricKeys](https://github.com/flyteorg/flyteadmin/blob/91787d2d087ccbc21a0a2e99f1826ca814cc232f/cmd/entrypoints/serve.go#L50) in order for your new context values to be[ added as metric labels](https://github.com/flyteorg/flytestdlib/blob/15cac7bec8470974050ed02343cda8191e5f14bc/promutils/labeled/counter.go#L23)

Reply:
cc @EngHabu

Reply:
makes sense. Moved `ErrorKindKey` const to common and added to [metric keys](https://github.com/alexapdev/flyteadmin/blob/patch-1/cmd/entrypoints/serve.go#L50)